tests/test_rag_embeddings.py

```
@@ -0,0 +1,392 @@
+    def test_get_embeddings_batch_empty_input(self, mock_configure, mock_get_api_key):
+        """Test batch embedding with empty input list."""
+        # Execute function with empty list
+        result = get_embeddings_batch([])
+
+        # Verify empty list is returned
+        assert result == []
+
+    @patch('src.rag.embeddings.get_google_api_key')
+    @patch('src.rag.embeddings.genai.configure')
+    @patch('src.rag.embeddings.genai.batch_embed_contents')
+    def test_get_embeddings_batch_with_different_task_type(self, mock_batch_embed, mock_configure, mock_get_api_key):
```

Ruff (0.14.8)

Unused method argument: `mock_configure`

(ARG002)